tests/e2e/framework/kubernetes.go

```diff
@@ -793,14 +800,27 @@ EOF`, k.KubeConfig, dummyValidationRule)
 	return nil
 }
 
-func (k *KubeInfo) deployIstioWithHelm() error {
-	yamlFileName := filepath.Join(istioInstallDir, helmInstallerName, "istio", "templates", "crds.yaml")
+func (k *KubeInfo) deployCRDs(kubernetesCRD string) error {
+	yamlFileName := filepath.Join(istioInstallDir, helmInstallerName, "istio-init", "files", kubernetesCRD)
 	yamlFileName = filepath.Join(k.ReleaseDir, yamlFileName)
 
 	// deploy CRDs first
 	if err := util.KubeApply("kube-system", yamlFileName, k.KubeConfig); err != nil {
-		log.Errorf("Failed to apply %s", yamlFileName)
-		return err
+		return fmt.Errorf("failed to apply %s because %v", yamlFileName, err)
+	}
+	return nil
+}
+
+func (k *KubeInfo) deployIstioWithHelm() error {
+	// Note: When adding a CRD to the install, a new CRDFile* constant is needed
+	// This slice contains the list of CRD files installed during testing
+	istioCRDFileNames := []string{CRDZeroInstallFile, CRDOneInstallFile, CRDTwoInstallFile}
+
+	// deploy all CRDs in Istio first
+	for _, yamlFileName := range istioCRDFileNames {
+		if err := k.deployCRDs(yamlFileName); err != nil {
+			return fmt.Errorf("failed to apply all Istio CRDs: %v", err)
```

Comment:
Please add file name to the message, something like this:
`return fmt.Errorf("failed to apply all Istio CRDs from file: %s with error: %v", yamlFileName, err)`
In this it will be easier to troubleshoot as we know the exact file name which cause the failure.

Reply:
thanks will do.